hypershift-operator/controllers/util/util.go

```diff
@@ -0,0 +1,39 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func ParseNamespacedName(name string) types.NamespacedName {
+	parts := strings.SplitN(name, string(types.Separator), 2)
+	if len(parts) > 1 {
+		return types.NamespacedName{Namespace: parts[0], Name: parts[1]}
+	}
+	return types.NamespacedName{Name: parts[0]}
+}
+
+func NamespacedName(obj metav1.Object) types.NamespacedName {
```

Comment:
This is the same as `client.ObjectKeyFromObject` and can be deleted/replaced with usage of the existing client function

Reply:
Dropped `NamespacedName`